File: jmillord-accelerated-restaurant/src/entities/Order.ts
import { Client } from './Client'
import { Cashier } from './Cashier'
import { OrderDetails } from './OrderDetail'
import { Md5 } from 'ts-md5/dist/md5'

export class Order {
  private id: string;
  private transactionDate: Date
  private client: Client;
  private cashier: Cashier;
  private orderDetails: OrderDetails
  private totalAmount: number;

  constructor() { }

  public getId(): string {
    return this.id;
  }

  public setId(id: string): void {
    this.id = id;
  }

  public getTransactionDate(): Date {
    return this.transactionDate;
  }

  public setTransactionDate(transactionDate: Date): void {
    this.transactionDate = transactionDate;
  }

  public getClient(): Client {
    return this.client;
  }

  public setClient(client: Client): void {
    this.client = client;
  }

  public getCashier(): Cashier {
    return this.cashier;
  }

  public setCashier(cashier: Cashier): void {
    this.cashier = cashier;
  }

  public getOrderDetails(): OrderDetails {
    return this.orderDetails;
  }

  public setOrderDetails(orderDetails: OrderDetails): void {
    this.orderDetails = orderDetails;
  }

  public getTotalAmount(): number {
    return this.totalAmount;
  }

  public setTotalAmount(totalAmount: number): void {
    this.totalAmount = totalAmount;
  }

  public hashCode() {
    return Md5.hashStr(this.id)
  }



}

File: monegros-restaurant-dianamonegrodiaz/src/monegros/restaurant/entities/Order.java
package monegros.restaurant.entities;

import java.time.Instant;
import java.util.List;
import java.util.Objects;

public class Order {
    private Long id;
    private Chef chef;
    private Instant transactionDate;
    private Client client;
    private Cashier cashier;
    private List<OrderDetail> orderDetails;
    private Double totalAmount;
    private String status;

    public Order() {
    	
    }

    public Order(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
    public Chef getChef() {
		return chef;
	}

	public void setChef(Chef chef) {
		this.chef = chef;
	}

    public Instant getTransactionDate() {
        return transactionDate;
    }

    public void setTransactionDate(Instant transactionDate) {
        this.transactionDate = transactionDate;
    }

    public Client getClient() {
        return client;
    }

    public void setClient(Client client) {
        this.client = client;
    }

    public Cashier getCashier() {
        return cashier;
    }

    public void setCashier(Cashier cashier) {
        this.cashier = cashier;
    }

    public List<OrderDetail> getOrderDetails() {
        return orderDetails;
    }

    public void setOrderDetails(List<OrderDetail> orderDetails) {
        this.orderDetails = orderDetails;
     }

    public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}

	public Double getTotalAmount() {
        return totalAmount;
    }

	public void setTotalAmount(Double totalAmount) {
        this.totalAmount = totalAmount;
    }
    

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Order order = (Order) o;
        return Objects.equals(id, order.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

	
}
